### PR TITLE
Adding initial IP and User Agent blacklist support

### PR DESCRIPTION
**Pull request by brian2000**

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -150,6 +150,22 @@ function wbz404_pluginActivation() {
...
+  $wpdb->query($query);
+
+  $query = "CREATE TABLE IF NOT EXISTS `" . $wpdb->prefix . "wbz404_ignored_bots` (
+    `ID` bigint(40) NOT NULL auto_increment,
+    `user_agent` varchar(255) NOT NULL,
+    PRIMARY KEY  (`ID`),
+    KEY `user_agent` (`user_agent`)
+  ) ENGINE=MyISAM " . $charset_collate . " COMMENT='404 Redirected Plugin ignored IP bots' AUTO_INCREMENT=1";
+  $wpdb->query($query);
+
```
Comment (pwlee) [inline includes/functions.php:168]: Super nit-picky, but let's match the indentation of the file.  (Turn these spaces into tabs)

File: includes/functions.php [hunk 2]
```diff
@@ -527,3 +543,53 @@ function wbz404_ProcessRedirect($redirect) {
...
+  if(!empty($_SERVER['HTTP_X_CLUSTER_CLIENT_IP']))
+    return $_SERVER['HTTP_X_CLUSTER_CLIENT_IP'];
+  if(!empty($_SERVER['HTTP_FORWARDED_FOR']))
+    return $_SERVER['HTTP_FORWARDED_FOR'];
+  if(!empty($_SERVER['HTTP_FORWARDED']))
+    return $_SERVER['HTTP_FORWARDED'];
+
+  # return REMOTE_ADDR since all else failed
+  return $_SERVER['REMOTE_ADDR'];
+}
```
Comment (pwlee) [inline includes/functions.php]: Super nit-picky, but let's match the indentation of the file.  (Turn these spaces into tabs)

**Comment by pwlee** [comment]
Just some very minor indentation stuff.  Otherwise, this looks really good, man.  Great work.

**Comment by brian2000** [comment]
After we reviewed the original code we have decided to indent with spaces (2 spaces per tab) as the author was using a mixture of both tabs and spaces. It makes sense here to convert the file's indentation to spaces (our prefered indentation strategy) and move forward.

**Pull request merged**